ma_gym/envs/traffic_junction/traffic_junction.py

```diff
@@ -125,25 +271,171 @@ def __create_grid(self):
 
         return _grid
 
+    def step(self, agents_action):
+        """
+        Performs an action in the environment and steps forward. At each step a new agent enters the road by
+        one of the 4 gates according to a probability "_arrive_prob". A "ncoll" reward is given to an agent if it
+        collides and all of them receive "-0.01*step_n" to avoid traffic jams.
+
+        :param agents_action: list of actions of all the agents to perform in the environment
+        :type agents_action: list
+
+        :return: agents observations, rewards, if agents are done and additional info
+        :rtype: tuple
+        """
+        assert len(agents_action) == self.n_agents, \
+            "Invalid action! It was expected to be list of {}" \
+            " dimension but was found to be of {}".format(self.n_agents, len(agents_action))
+
+        # Todo: Add check to validate each sub-action. It should be a valid action.
+        # This must be done before position of any car is updated
```

Comment:
Add a check to validate each sub-action before updating any of the agent's position.

As of now, it is being checked in `__update_agent_pos`, but that checking it over there would lead to position updates to some agents.